src/OpenTracing/Mock/MockSpan.cs

```diff
@@ -298,8 +298,28 @@ public struct Reference
 
             public Reference(MockSpanContext context, string referenceType)
             {
-                Context = context;
-                ReferenceType = referenceType;
+                Context = context ?? throw new ArgumentNullException(nameof(context));
+                ReferenceType = referenceType ?? throw new ArgumentNullException(nameof(referenceType));
+            }
+
+            public override bool Equals(object obj)
+            {
+                return Equals(obj as Reference);
+            }
+
+            public bool Equals(Reference other)
+            {
+                return other != null &&
+                       EqualityComparer<MockSpanContext>.Default.Equals(Context, other.Context) &&
```

Comment:
`EqualityComparer<MockSpanContext>.Default` wow, VS generates weird Equality. They should take a hint from R# :-)

[not asking for a change here]